pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -79,12 +79,13 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *pav1alpha1.PodAutosc
 	}
 
 	// Having an SKS and its PrivateServiceName is a prerequisite for all upcoming steps.
-	if sks == nil || (sks != nil && sks.Status.PrivateServiceName == "") {
+	if sks == nil || sks.Status.PrivateServiceName == "" {
 		// Before we can reconcile decider and get real number of activators
 		// we start with default of 2.
 		if _, err = c.ReconcileSKS(ctx, pa, nv1alpha1.SKSOperationModeServe, 0 /*numActivators == all*/); err != nil {
 			return fmt.Errorf("error reconciling SKS: %w", err)
 		}
+		pa.Status.MarkDependenciesNotReady("DependenciesNotReady", "SKS is provisioning")
```

Comment:
"DependenciesNotReady" is redundant with the new Type.  I'd use this to identify the dependency itself, e.g. SKSNotReady or something